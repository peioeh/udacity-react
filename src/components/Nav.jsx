import { NavLink } from "react-router-dom";
import User from "./User";


const Nav = () => {
  return (
    <nav className="nav">
      <ul>
        <li>
          <NavLink to="/" className={({ isActive }) => (isActive ? "active" : "")}>Home</NavLink>
        </li>
        <li>
          <NavLink to="/leaderboard" className={({ isActive }) => (isActive ? "active" : "")}>Leaderboard</NavLink>
        </li>
        <li>
          <NavLink to="/add" className={({ isActive }) => (isActive ? "active" : "")}>New</NavLink>
        </li>
        <li className="nav-right">
          <User />
        </li>
        <li>
          <NavLink to="/logout" className={({ isActive }) => (isActive ? "active" : "")}>Logout</NavLink>
        </li>
      </ul >
    </nav >
  );
};

export default Nav;