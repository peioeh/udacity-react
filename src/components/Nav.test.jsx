/**
 * @jest-environment jsdom
 */
import "@inrupt/jest-jsdom-polyfills"
import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react';
import { fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import { legacy_createStore as createStore } from 'redux';
import reducer from "../reducers";
import middleware from "../middleware";

import Nav from './Nav';


const store = createStore(reducer, middleware);

test('nav_bar_snapshot_test', () => {
  render(<MemoryRouter><Provider store={store}><Nav /></Provider></MemoryRouter>);
  //screen.debug();
  expect(screen.getByRole('navigation')).toMatchSnapshot();
});



test('nav_bar_click_test', () => {
  render(<MemoryRouter><Provider store={store}><Nav /></Provider></MemoryRouter>);

  const navLink = screen.getByText('Home');
  fireEvent.click(navLink);

  expect(screen.getByText('Home')).toHaveClass('active');
});