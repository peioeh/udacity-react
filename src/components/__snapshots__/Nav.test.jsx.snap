// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav_bar_snapshot_test 1`] = `
<nav
  class="nav"
>
  <ul>
    <li>
      <a
        aria-current="page"
        class="active"
        data-discover="true"
        href="/"
      >
        Home
      </a>
    </li>
    <li>
      <a
        class=""
        data-discover="true"
        href="/leaderboard"
      >
        Leaderboard
      </a>
    </li>
    <li>
      <a
        class=""
        data-discover="true"
        href="/add"
      >
        New
      </a>
    </li>
    <li
      class="nav-right"
    >
      <div />
    </li>
    <li>
      <a
        class=""
        data-discover="true"
        href="/logout"
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;
